File: Gestion Atelier De Couture/src/entity/Article.java
package entity;

import services.Sys;

public class Article {

    protected String id;
    protected String nomArticle;
    protected double prix;
    protected int quantiteStock;
    protected double montantStock;
    protected String type;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNomArticle() {
        return nomArticle;
    }

    public void setNomArticle(String nomArticle) {
        this.nomArticle = nomArticle;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getQuantite() {
        return quantiteStock;
    }

    public void setQuantite(int quantite) {
        this.quantiteStock = quantite;
    }

    public double getMontant() {
        return montantStock;
    }

    public void setMontant(double montant) {
        this.montantStock = montant;
    }
    public void copy(Article a) {
        this.id = a.getId();
        this.nomArticle = a.getNomArticle();
        this.prix = a.getPrix();
        this.quantiteStock = a.getQuantite();
        this.montantStock = a.getMontant();
        this.type = a.getType();
    }

    public boolean equals(Article a) {
        return this.nomArticle.equals(a.getNomArticle()) && 
               this.prix == a.getPrix();
    }

    public String toString() {
        return "\n---------------------------------------\n" +
                "ID          : " + id + "\n" +
                "Nom Article : " + Sys.capitalize(nomArticle) + "\n" +
                "Prix        : " + prix + " Fcfa" +"\n" +
                "Quantite    : " + quantiteStock + "\n" +
                "Montant     : " + montantStock + " Fcfa" + "\n" +
                "Type        : " + type;
    }
    
}

File: Gestion A C/models/Article.php
<?php
require_once 'Sys.php';

class Article {
    protected string $id;
    protected string $nomArticle;
    protected float $prix;
    protected int $quantiteStock;
    protected float $montantStock;
    protected string $type;

    // Getters et Setters
    public function getId(): string {
        return $this->id;
    }

    public function setId(string $id): void {
        $this->id = $id;
    }

    public function getNomArticle(): string {
        return $this->nomArticle;
    }

    public function setNomArticle(string $nomArticle): void {
        $this->nomArticle = $nomArticle;
    }

    public function getPrix(): float {
        return $this->prix;
    }

    public function setPrix(float $prix): void {
        $this->prix = $prix;
    }

    public function getQuantite(): int {
        return $this->quantiteStock;
    }

    public function setQuantite(int $quantite): void {
        $this->quantiteStock = $quantite;
    }

    public function getMontant(): float {
        return $this->montantStock;
    }

    public function setMontant(float $montant): void {
        $this->montantStock = $montant;
    }

    public function getType(): string {
        return $this->type;
    }

    public function setType(string $type): void {
        $this->type = $type;
    }

    // Copie les données d’un autre article
    public function copy(Article $a): void {
        $this->id = $a->getId();
        $this->nomArticle = $a->getNomArticle();
        $this->prix = $a->getPrix();
        $this->quantiteStock = $a->getQuantite();
        $this->montantStock = $a->getMontant();
        $this->type = $a->getType();
    }

    // Comparaison d’articles
    public function equals(Article $a): bool {
        return $this->nomArticle === $a->getNomArticle() &&
               $this->prix === $a->getPrix();
    }

    public function __toString(): string {
        return "\n---------------------------------------\n" .
            "ID          : " . $this->id . "\n" .
            "Nom Article : " . Sys::capitalize($this->nomArticle) . "\n" .
            "Prix        : " . $this->prix . " Fcfa\n" .
            "Quantite    : " . $this->quantiteStock . "\n" .
            "Montant     : " . $this->montantStock . " Fcfa\n" .
            "Type        : " . $this->type;
    }
}
